components/eamxx/src/share/field/field_utils_impl.hpp

```diff
@@ -594,6 +594,244 @@ ST field_min(const Field& f, const ekat::Comm* comm)
+  //   - along 1st dimension
+  //   - along 2nd dimension if rank>2
+  // So if the input field is 2d and the index to subview at is along
+  // the 2nd dim, we must stop recursion, and print the field manually
+  // extracting the last slice.
+  //
+  // We keep all the tags/indices we slice away, since we need them at
+  // the end of recursion when we print the info of the field location.
+  // E.g., if f has tags/dims <COL,CMP,LEV>/(2,3,4) and we subview at
+  // tags/indices <COL,LEV>/(0,1), we want to print something like
+  //   f(0,:,1):
+  //     0.123, 0.456, 0.789
+
+  EKAT_REQUIRE_MSG (tags.size()==indices.size(),
+      "Error! Tags vector size differs from indices vector size.\n");
+
+  const auto& layout = f.get_header().get_identifier().get_layout();
+
+  // Get layout of original field (before all the slicing happened)
+  auto get_orig_header = [&]() -> std::shared_ptr<const FieldHeader> {
+    auto fh = f.get_header_ptr();
+    while (fh->get_identifier().get_layout().rank()<orig_rank) {
+      fh = fh->get_parent().lock();
+    }
+    return fh;
+  };
+
+  // Get a vector of strings, which we'll use to print the info of the field
+  // slice/entry currently printed. We can already fill the entries corresponding
+  // to dimensions we sliced away.
+  auto get_dims_str = [&](const FieldLayout& orig_layout) -> std::vector<std::string> {
+    const int orig_rank = orig_layout.rank();
+    const int num_tags  = tags.size();
+    std::vector<std::string> dims_str (orig_rank,"");
+    for (int ii=0, jj=0; ii<orig_rank && jj<num_tags; ++ii) {
+      if (orig_layout.tag(ii)==tags[jj]) {
+        // Was sliced. Store slice idx as a stirng
+        dims_str[ii] = std::to_string(indices[jj]);
+        ++jj;
+      }
+    }
+    return dims_str;
+  };
+
+  // Get the dimensions (in the original field) that were *NOT* sliced away,
+  // since we'll have to loop over them. We'll use these indices to add the
+  // missing info in the dims_str vector<string> as we loop and print.
+  auto get_dims_left = [&](const FieldLayout& orig_layout) -> std::vector<int> {
+    const int orig_rank = orig_layout.rank();
+    const int num_tags  = tags.size();
+    std::vector<int> dims_left;
+    for (int ii=0, jj=0; ii<orig_rank; ++ii) {
+      if (jj<num_tags && orig_layout.tag(ii)==tags[jj]) {
+        // Was sliced. Skip
+        ++jj;
+      } else {
+        // Was not sliced.
+        dims_left.push_back(ii);
+      }
+    }
+    return dims_left;
+  };
+
+  constexpr int max_per_line = 5;
+  if (curr_idx==tags.size()) {
+    // All slices have been taken. Print the whole input field.
+    const auto& orig_layout = get_orig_header()->get_identifier().get_layout();
+    const auto dims_left = get_dims_left(orig_layout);
+    auto dims_str = get_dims_str(orig_layout);
+
+    f.sync_to_host();
+    const int rank = layout.rank();
+    out << " -----------------------------------------------------------------------\n";
```

Comment:
See my comment about format of output in the "conversation" section.  I think there are two many `-----....` lines which make it difficult to parse the error message.